### PR TITLE
[TapoControl] Adding P300 to the list of supported equipments

### PR DESCRIPTION
**Pull request by clinique**

### CODE REVIEW

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoSubRequest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import static org.openhab.binding.tapocontrol.internal.constants.TapoBindingSettings.*;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * {@TapoSubRequest} hols data sent to device in order to act on a child
```
Comment (jlaur) [inline bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoSubRequest.java:24]:
```suggestion
 * {@TapoSubRequest} holds data sent to device in order to act on a child
```

File: bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/P300.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="tapocontrol"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- P115 THING-TYPE (SOCKET) -->
```
Comment (jlaur) [inline bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/P300.xml:7]:
```suggestion
	<!-- P300 THING-TYPE (POWER STRIP) -->
```

File: bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/channels.xml [hunk 3]
```diff
@@ -37,6 +37,22 @@
 		</channels>
 	</channel-group-type>
 
+	<channel-group-type id="powerStrip">
+		<label>SmartPlug</label>
+		<description>Tapo Smart Plug Power Outlet</description>
+		<channels>
+			<channel id="output1" typeId="outputChannel">
+				<label>Output Switch #1</label>
```
Comment (jlaur) [inline bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/channels.xml:45]: Consider also providing (overriding) descriptions mentioning 1, 2 and 3. Is "#" needed here, or simply:
```suggestion
				<label>Output Switch 1</label>
```

?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for adding the P300! I have provided feedback in the form of some minor comments.

File: bundles/org.openhab.binding.tapocontrol/README.md [hunk 4]
```diff
@@ -19,6 +19,7 @@ The following Tapo-Devices are supported. For precise channel-description look a
 | MultiColor LightStrip (Wi-Fi)      | L900        | Multicolor RGB Dimmable LightStrip (5m)     |
 |                                    | L920        | Multicolor RGB-IC ColorZone LightStrip (5m) |
 |                                    | L930        | Multicolor RGBW-IC 50-Zone LightStrip (5m)  |
+| Power Strip (Wi-Fi)                | P300        | Smart Wi-Fi Power Strip - 3 sockets         |
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/README.md:22]: Done

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/TapoDeviceConnector.java [hunk 5]
```diff
@@ -159,6 +158,27 @@ public void sendDeviceCommand(String name, Object value) {
...
+        long now = System.currentTimeMillis();
+        if (now > this.lastSent + TAPO_SEND_MIN_GAP_MS) {
+            this.lastSent = now;
+            getChild(index).ifPresent(child -> {
+                child.setDeviceOn(Boolean.valueOf((Boolean) value));
+                TapoSubRequest request = new TapoSubRequest(child.getDeviceId(), DEVICE_CMD_SETINFO, child);
+                sendSecurePasstrhroug(GSON.toJson(request), request.method());
+            });
+        } else {
+            logger.debug("({}) command not sent becauso of min_gap: {}", uid, now + " <- " + lastSent);
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/TapoDeviceConnector.java:178]: Corrected

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/TapoDeviceConnector.java [hunk 6]
```diff
@@ -212,6 +233,23 @@ public void queryInfo(boolean ignoreGap) {
         }
     }
 
+    /**
+     * Query Info from Child Devices and refresh deviceInfo
+     *
+     * @param ignoreGap ignore gap to last query. query anyway
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/TapoDeviceConnector.java:239]: Removed

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/constants/TapoThingConstants.java [hunk 7]
```diff
@@ -85,6 +88,9 @@ public class TapoThingConstants {
     /*** THINGS WITH ENERGY DATA ***/
     public static final Set<ThingTypeUID> SUPPORTED_ENERGY_DATA_UIDS = Set.of(P110_THING_TYPE, P115_THING_TYPE);
 
+    /*** THINGS WITH CHIDS DATA ***/
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/constants/TapoThingConstants.java:91]: Corrected

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoSubRequest.java (hunk 1 above)
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/structures/TapoSubRequest.java:24]: Corrected

File: bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/P300.xml (hunk 2 above)
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/P300.xml:7]: Modified

File: bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/channels.xml (hunk 3 above)
Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/channels.xml:45]: Done

Comment (jlaur) [reply bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/channels.xml]: Please regenerate I18N after this change.

Comment (clinique) [reply bundles/org.openhab.binding.tapocontrol/src/main/resources/OH-INF/thing/channels.xml]: Done.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**